src/grids.jl

```diff
@@ -475,9 +694,15 @@ function clenshaw_curtis_weights(nlat_half::Integer)
     θs = get_colat(FullClenshawGrid,nlat_half)
     return [4sin(θj)/(nlat+1)*sum([sin(p*θj)/p for p in 1:2:nlat]) for θj in θs[1:nlat_half]]
 end
+healpix_weights(nside::Integer) =
+    [nlon_healpix(nside,j) for j in 1:2nside].*(2/npoints_healpix(nside))
+healpix4_weights(nside::Integer) =
+    [nlon_healpix4(nside,j) for j in 1:nside].*(2/npoints_healpix4(nside))
 
 # SOLID ANGLES ΔΩ = sinθ Δθ Δϕ
 get_solid_angles(Grid::Type{<:AbstractGrid},nlat_half::Integer) = 
     get_quadrature_weights(Grid,nlat_half) .* (2π./get_nlons(Grid,nlat_half))
 get_solid_angles(Grid::Type{<:HEALPixGrid},nside::Integer) =
     4π/get_npoints(Grid,nside)*ones(get_nlat_half(Grid,nside))
+get_solid_angles(Grid::Type{<:HEALPix4Grid},nside::Integer) =
+    4π/get_npoints(Grid,nside)*ones(get_nlat_half(Grid,nside))
```

Comment:
We could just `Union{HEALPixGrid,HEALPix4Grid}` here, no? I believe in general it might make sense to add a bit more structure into the grid type hierarchy, what about

abstract type AbstractGrid <: AbstractVector end
abstract type AbstractHEALPixGrid <: AbstractGrid end
abstract type AbstractHEALPix12Grid <: AbstractHEALPixGrid end
abstract type AbstractHEALPix4Grid <: AbstractHEALPixGrid end
```
and the concrete types as subtypes of those?

Reply:
Agreed. Let's refactor the code like this when we add another HEALPix-type grid.